File: Java/OPSJLib/src/ops/OPSConfig.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ops;

import configlib.ArchiverInOut;
import configlib.XMLArchiverIn;
import configlib.exception.FormatException;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.Vector;
import ops.archiver.OPSObjectFactory;

/**
 *
 * @author angr
 */
public class OPSConfig extends OPSObject
{
    //Static------------------------

    //static OPSConfig theConfig = null;
    static OPSConfig getConfig() throws IOException, FormatException
    {

        FileInputStream fis = new FileInputStream("ops_config.xml");
        XMLArchiverIn archiverIn = new XMLArchiverIn(fis, "root");
        archiverIn.add(OPSObjectFactory.getInstance());

        OPSConfig newConfig = null;
        newConfig = (OPSConfig) archiverIn.inout("ops_config", newConfig);

        return newConfig;
    }

    static OPSConfig getConfig(File configFile) throws IOException, FormatException
    {

        FileInputStream fis = new FileInputStream(configFile);
        XMLArchiverIn archiverIn = new XMLArchiverIn(fis, "root");
        archiverIn.add(OPSObjectFactory.getInstance());
        OPSConfig newConfig = null;
        newConfig = (OPSConfig) archiverIn.inout("ops_config", newConfig);

        return newConfig;
    }
    //------------------------------
    private Vector<Domain> domains = new Vector<Domain>();

    public OPSConfig()
    {
        appendType("OPSConfig");
    }

    public Domain getDomain(String domainID)
    {
        for (Domain domain : domains)
        {
            if (domain.getDomainID().equals(domainID))
            {
                return domain;
            }
        }
        return null;
    }

    @Override
    public void serialize(ArchiverInOut archive) throws IOException
    {
        super.serialize(archive);
        domains = (Vector<Domain>) archive.inoutSerializableList("domains", domains);
    }
}

File: trunk/C#/Ops/OPSConfig.cs
///////////////////////////////////////////////////////////
//  OPSConfig.cs
//  Implementation of the Class OPSConfig
//  Created on:      12-nov-2011 09:25:33
//  Author:
///////////////////////////////////////////////////////////

using System.Collections.Generic;       // Needed for the "List"
using System.IO;
using System.Text;

namespace Ops 
{
	public class OPSConfig : OPSObject 
    {
        private List<Domain> domains = new List<Domain>();

        public static OPSConfig GetConfig() 
        {
            return GetConfig("ops_config.xml");
        }

        public static OPSConfig GetConfig(string configFile)
        {
            FileStream fis = File.OpenRead(configFile);
            XMLArchiverIn archiverIn = new XMLArchiverIn(fis, "root");
            archiverIn.Add(OPSObjectFactory.GetInstance());
            OPSConfig newConfig = null;
            newConfig = (OPSConfig)archiverIn.Inout("ops_config", newConfig);
            fis.Close();
            return newConfig;
        }

        public static void SaveConfig(OPSConfig conf, string configFile) 
        {
            FileStream fos = File.OpenWrite(configFile);
            BinaryWriter bw = new BinaryWriter(fos);
            XMLArchiverOut archiverOut = new XMLArchiverOut(bw, false);
            bw.Write(Encoding.ASCII.GetBytes("<root xmlns=ops>\n"));
            archiverOut.Inout("ops_config", conf);
            bw.Write(Encoding.ASCII.GetBytes("</root>"));
            fos.Close();
        }

        //---------------------------------------------------------------

        public OPSConfig()
        {
            AppendType("OPSConfig");
        }

        public Domain GetDomain(string domainID)
        {
            foreach (Domain domain in domains)
            {
                if (domain.GetDomainID().Equals(domainID))
                {
                    return domain;
                }
            }
            return null;
        }

        public override void Serialize(IArchiverInOut archive)
        {
            base.Serialize(archive);
            domains = (List<Domain>)archive.InoutSerializableList("domains", domains);
        }

        public List<Domain> getDomains()
        {
            return domains;
        }

	}

}